google/cloud/spanner/samples/samples.cc

```diff
@@ -666,6 +666,110 @@ void GetDatabaseDdl(google::cloud::spanner_admin::DatabaseAdminClient client,
+  namespace spanner = ::google::cloud::spanner;
+  auto client = spanner::Client(spanner::MakeConnection(
+      spanner::Database(project_id, instance_id, database_id),
+      google::cloud::Options{}.set<spanner::SessionCreatorRoleOption>(role)));
+  spanner::SqlStatement select_star("SELECT * FROM Singers");
+  auto rows = client.ExecuteQuery(std::move(select_star));
+  using RowType =
+      std::tuple<std::int64_t, std::string, std::string, spanner::Bytes>;
+  for (auto const& row : spanner::StreamOf<RowType>(rows)) {
+    if (!row) throw std::runtime_error(row.status().message());
+    std::cout << "SingerId: " << std::get<0>(*row) << ", "
+              << "FirstName: " << std::get<1>(*row) << ", "
+              << "LastName: " << std::get<2>(*row) << "\n";
+  }
+}
+//! [END spanner_read_data_with_database_role]
+
+//! [START spanner_list_database_roles]
+void ListDatabaseRoles(google::cloud::spanner_admin::DatabaseAdminClient client,
+                       std::string const& project_id,
+                       std::string const& instance_id,
+                       std::string const& database_id) {
+  google::cloud::spanner::Database database(project_id, instance_id,
+                                            database_id);
+  std::cout << "Database Roles are:\n";
+  for (auto const& role : client.ListDatabaseRoles(database.FullName())) {
+    if (!role) throw std::runtime_error(role.status().message());
+    std::cout << role->name() << "\n";
+  }
+}
+//! [END spanner_list_database_roles]
+
+//! [START spanner_enable_fine_grained_access]
+void EnableFineGrainedAccess(
+    google::cloud::spanner_admin::DatabaseAdminClient client,
+    std::string const& project_id, std::string const& instance_id,
+    std::string const& database_id, std::string const& iam_member,
+    std::string const& database_role, std::string const& title) {
+  google::cloud::spanner::Database database(project_id, instance_id,
+                                            database_id);
+
+  google::iam::v1::GetIamPolicyRequest request;
+  request.set_resource(database.FullName());
+  request.mutable_options()->set_requested_policy_version(3);
+  auto policy = client.GetIamPolicy(request);
+  if (!policy) throw std::runtime_error(policy.status().message());
+  if (policy->version() < 3) policy->set_version(3);
+
+  auto& binding = *policy->add_bindings();
+  binding.set_role("roles/spanner.fineGrainedAccessUser");
+  binding.add_members(iam_member);
+  auto& condition = *binding.mutable_condition();
+  condition.set_expression("resource.name.endsWith(\"/databaseRoles/" +
+                           database_role + "\")");
+  condition.set_title(title);
+
+  auto new_policy =
+      client.SetIamPolicy(database.FullName(), *std::move(policy));
+  if (!new_policy) throw std::runtime_error(new_policy.status().message());
+  std::cout << "Enabled fine-grained access in IAM. New policy has version "
+            << new_policy->version() << "\n";
```

Comment:
I do not think that version is what you think it is.  It is more like a code revision tracker than a state change tracker.  You want something like the `etag` field.

Reply:
I don't think I think it is anything in particular.  It doesn't seem, at least, that the sample implies that it isn't a code-revision tracker.

That said, the shape of the output is defined by the internal "Fine Grained Access Control Samples" document, which is followed by all the other languages, so I'm not sure I'm in a position to change anything.

Reply:
You could try to talk the team into changing this example. As it is, it will print either `1` or `3` for the version.

Reply:
Please merge as-is though, we can implement (or not) both my suggestions at a later time.

Reply:
> You could try to talk the team into changing this example.

I can certainly do that.  I'm not sure what I would write other than your comment verbatim however, so I'm also happy to point you at the doc.  Let me know offline.

> As it is, it will print either 1 or 3 for the version.

I thought it would always print (at least) 3, given that's what we raised the minimum version to above.